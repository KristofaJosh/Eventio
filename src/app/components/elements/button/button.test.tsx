import React from "react";
import renderer from "react-test-renderer";
import Button, { RoundButton } from ".";

it("Button renders correctly", () => {
    const tree = renderer.create(<Button text="click me" />).toJSON();
    const danger = renderer.create(<Button btnType="danger" text="click me" />).toJSON();
    const soft = renderer.create(<Button btnType="soft" text="click me" />).toJSON();
    const smallBtn = renderer.create(<Button size="small" text="click me" />).toJSON();

    expect(tree).toMatchSnapshot();
    expect(danger).toMatchSnapshot();
    expect(soft).toMatchSnapshot();
    expect(smallBtn).toMatchSnapshot();
});

it("Round Button renders correctly", () => {
    const tree = renderer.create(<RoundButton />).toJSON();
    expect(tree).toMatchSnapshot();
});
