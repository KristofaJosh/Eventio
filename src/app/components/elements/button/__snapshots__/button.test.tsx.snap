// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
<button
  aria-label="click me"
  className="button button__isNormal button__btnPrimary   "
>
  click me
</button>
`;

exports[`Button renders correctly 2`] = `
<button
  aria-label="click me"
  className="button button__isNormal button__btnDanger   "
>
  click me
</button>
`;

exports[`Button renders correctly 3`] = `
<button
  aria-label="click me"
  className="button button__isNormal button__btnSoft   "
>
  click me
</button>
`;

exports[`Button renders correctly 4`] = `
<button
  aria-label="click me"
  className="button button__isSmall button__btnPrimary   "
>
  click me
</button>
`;

exports[`Round Button renders correctly 1`] = `
<button
  className="button button__isRound  "
  type="button"
/>
`;
